docs/activities.md

```diff
@@ -39,6 +39,22 @@ Activity Definitions are named and referenced in code by their <a class="tdlp" h
 
 <div class="tdiw"><div class="tditw"><p class="tdit">Activity Definition</p></div><div class="tdiiw"><img class="tdi" src="/diagrams/activity-definition.svg" alt="Activity Definition" /></div></div>
 
+#### Idempotency
+
+Temporal recommends that Activities are idempotent.
+
+Activities are idempotent if multiple applications of that operation do not change the state of the system beyond the initial application.
+
+Activities may be retried repeatedly, so you may need to use idempotency keys for critical side effects.
+
+Unlike determinism in a Workflow, which actually is a requirement, Temporal cannot enforce that your Activity is idempotent.
+
+The lack of idempotency may affect the correctness of your application, but does not affect Temporal (in other words, it does not lead to a platform error).
+
+<!-- Temporal supports either at most once or at least once execution of Activities, and in the case of at least once activities must be idempotent -->
+
+In some cases, whether something is idempotent does not affect the correctness of an application. For example, if you have a monotonically-incrementing counter, you may not care that retries increment the counter because you don’t care about the actual value, only that the current value is greater than a previous value.
```

Comment:
Then we may need to consider another word besides "idempotent" such as "reentrant" or we should call it "idempotent from a Temporal POV" or something. Whether something actually affects other parts of the user system is unimportant to Temporal's note that all activities may be run multiple times.

We don't care about the user's side, we just need them to know an activity can run multiple times. Maybe "idempotency" isn't the phrasing we should use. Maybe we should just tell people that all activities may run multiple times in an undetectable way and they should be prepared for that.